website/docs/advanced-topics/struct-components/refs.mdx

```diff
@@ -3,53 +3,94 @@ title: 'Refs'
+are more modular. If you want to expose specific elements, for example the `<input>` element of a styled button, you should
+add a property of type `NodeRef` and bind the ref you receive from your components' parent to an element in the `html!`.
+
+A full example can be given by an externally controlled "light switch". Note that the state of the light
+is contained inside the `Light` component, and the `Room` ties this together by sending a message to the
+`Light` in the callback passed to the `Switch`.
 
 ```rust
-use web_sys::Element;
-use yew::{html, Component, Context, Html, NodeRef};
+use yew::prelude::*;
+use yew::html::{Scope, BindableRef};
+
+#[derive(PartialEq, Properties)]
+struct SwitchProps {
+    on_toggle: Callback<MouseEvent>,
+}
+
+#[function_component]
+fn Switch(SwitchProps { on_toggle }: &SwitchProps) -> Html {
+    html! {
+        <button onclick={on_toggle}>{"Switch"}</button>
+    }
+}
+
+enum LightMessage {
+    Toggle,
+}
 
-struct Comp {
-    node_ref: NodeRef,
+struct Light {
+    is_on: bool,
 }
 
-impl Component for Comp {
-    type Message = ();
+impl BaseComponent for Light {
     type Properties = ();
+    type Message = LightMessage;
+    type Reference = Scope<Self>;
 
-    fn create(_ctx: &Context<Self>) -> Self {
-        Self {
-            // highlight-next-line
-            node_ref: NodeRef::default(),
-        }
+    fn create(ctx: &Context<Self>, bindable_ref: BindableRef<Self::Reference>) -> Self {
+        bindable_ref.bind(ctx.link().clone());
+        Light { is_on: true }
     }
 
-    fn view(&self, _ctx: &Context<Self>) -> Html {
-        html! {
-            // highlight-next-line
-            <div ref={self.node_ref.clone()}></div>
-        }
+    fn update(&mut self, _ctx: &Context<Self>, LightMessage::Toggle: LightMessage) -> bool {
+        self.is_on = !self.is_on;
+        true
     }
 
-    fn rendered(&mut self, _ctx: &Context<Self>, _first_render: bool) {
-        // highlight-start
-        let has_attributes = self.node_ref
-            .cast::<Element>()
-            .unwrap()
-            .has_attributes();
-        // highlight-end
+    fn view(&self, _ctx: &Context<Self>) -> HtmlResult {
+        Ok(html! {
+            <div class={classes!["light", self.is_on.then(|| "on")]} />
+        })
+    }
+}
+
+#[function_component]
+fn Room() -> Html {
+    // highlight-next-line
+    let light_ref: ComponentRef<Light> = use_html_ref();
+    let on_toggle = {
+        let light_ref = light_ref.clone();
+        Callback::from(move |_| {
+            // highlight-start
+            light_ref
+                .get()
+                .expect("a light to have rendered")
+                .send_message(LightMessage::Toggle)
+            // highlight-end
+        })
+    };
+    html! {
+        <>
+            // highlight-next-line
+            <Light ref={&light_ref} />
+            <Switch {on_toggle} />
+        </>
     }
 }
 ```
```

Comment:
It's better to link to example on GitHub then put all of that in a documentation page